src/charm.py

```diff
@@ -1507,21 +1507,22 @@ def update_config(self, is_creating_backup: bool = False) -> bool:
     def _validate_config_options(self) -> None:
         """Validates specific config options that need access to the database or to the TLS status."""
         if (
-            self.config.instance_default_text_search_config is not None
+            self.config.instance_default_text_search_config != "pg_catalog.simple"
             and self.config.instance_default_text_search_config
             not in self.postgresql.get_postgresql_text_search_configs()
```

Comment:
We should consider storing the current value in the peer data and only check if it is changed.